lib/repository.py

```diff
@@ -81,7 +82,39 @@ def flush(self):
     if graphical:
         splash.Hide()
         sys.stdout = sys.__stdout__
-    print "The retriever is up-to-date"
+    print "\nThe retriever is up-to-date"
+
+
+def update_progressbar(progress):
+    """Shows progressbar given progress in float
+    Example: 0.5 for 50%
+    """
+    # Modify this to change the length of the progress bar
+    barWidth = 20
+    try:
+        rows, columns = os.popen('stty size', 'r').read().split()
+        barWidth = int(columns) - 20
+        if(not (barWidth > 1)):
+            barWidth = 20
+    except:
+        pass
+    status = ""
+    if isinstance(progress, int):
+        progress = float(progress)
+    if not isinstance(progress, float):
+        progress = 0
+        status = "error: progress var must be float\r\n"
+    if progress < 0:
+        progress = 0
+        status = "Halt...\r\n"
+    if progress >= 1:
+        progress = 1
+        status = "Done...\r\n"
+    block = int(round(barWidth*progress))
+    text = "\r[{0}] {1:.2f}%".format("#"*block + "-"*(barWidth-block),
```

Comment:
@ghoshbishakh great work. just a few things to change.
I think we can add Downlaod progress ` text = "\rDownload Progress: [{0}] `